include/deal.II/base/config.h.in

```diff
@@ -37,6 +37,7 @@
 #cmakedefine DEAL_II_WITH_64BIT_INDICES
 #cmakedefine DEAL_II_WITH_ADOLC
 #cmakedefine DEAL_II_WITH_ARPACK
+#cmakedefine DEAL_II_WITH_ARBORX
```

Comment:
Since we'd be handling Kokkos as an external package, would it make sense to add a `DEAL_II_WITH_KOKKOS` as well?

Reply:
Sure I'll add it. It's not used right now but at least it's there if someone wants to use Kokkos later on.